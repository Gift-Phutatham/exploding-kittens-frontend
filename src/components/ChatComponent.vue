<template>
  <v-card class="mx-auto" max-width="250">
    <v-banner class="text-center text-subtitle-1 py-2"> Chat Box </v-banner>
    <v-card class="d-flex flex-column-reverse overflow-y-auto" height="300" variant="flat">
      <v-card-item class="text-justify text-caption">
        <div v-for="(chat, index) in chats" :key="index">
          <span class="font-weight-bold">{{ chat.name }}</span> :
          <div>{{ chat.message }}</div>
          <v-divider class="my-1" v-if="index != chats.length - 1"></v-divider>
        </div>
      </v-card-item>
    </v-card>
    <v-divider></v-divider>
    <v-form class="d-flex" @submit.prevent>
      <v-text-field
        v-model="message"
        label="Message"
        variant="solo"
        single-line
        hide-details
        density="compact"
        append-inner-icon="mdi-send"
        @click:append-inner="sendMessage"
      ></v-text-field>
    </v-form>
  </v-card>
</template>

<script lang="ts">
import { defineComponent } from 'vue';

export default defineComponent({
  name: 'ChatComponent',
  data: () => ({
    message: '',
    chats: [],
  }),
  methods: {
    sendMessage() {
      if (this.message) {
        console.log(this.message);
        this.message = '';
      }
    },
  },
});
</script>
