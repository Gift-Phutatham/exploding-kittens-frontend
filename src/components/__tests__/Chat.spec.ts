import { describe, it, expect} from 'vitest';
import { mount } from '@vue/test-utils';
import { createVuetify } from 'vuetify';
import ChatComponent from '../ChatComponent.vue';
import vuetify from '@/plugins/vuetify';

describe('ChatComponent', () => {
    const vuetify = createVuetify();
    const wrapper = mount(ChatComponent, {
        global: {
            plugins: [vuetify],
        },
    });

    it('render component', () => {
        expect(wrapper.exists()).toBe(true);
    });

    it('adds a new chat message', async () => {
        const message = wrapper.find('input[type="text"]');
        const send = wrapper.find('.v-input__icon--append-inner');

        await message.setValue('Hello');
        await send.trigger('click');
    

        expect(wrapper.vm.chats).to.deep.equal({
            name: 'Anonymous',
            message: 'Hello',
        });
    });


    // it('send a message to the chat box', async () => {
    //     const message = wrapper.find('input[type=text]');
    //     const send = wrapper.find('button');

    //     message.setValue('Hello');
    //     await send.trigger('click');

    //     expect(wrapper.findAll('.text-caption').length).toBe(1);
    //     expect(wrapper.find('.text-caption div').text()).toBe('Hello');
    // });
});