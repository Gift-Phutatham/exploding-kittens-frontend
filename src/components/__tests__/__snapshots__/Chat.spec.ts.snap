// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChatComponent > render component 1`] = `
"<div class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated\\" style=\\"width: 225px;\\">
  <!---->
  <div class=\\"v-card__loader\\">
    <div class=\\"v-progress-linear v-theme--light\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
      <!---->
      <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
      <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
        <div class=\\"v-progress-linear__indeterminate\\">
          <div class=\\"v-progress-linear__indeterminate long\\"></div>
          <div class=\\"v-progress-linear__indeterminate short\\"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div class=\\"v-banner v-banner--stacked v-banner--density-default v-theme--light text-center text-subtitle-1 py-2\\" role=\\"banner\\">
    <!---->
    <div class=\\"v-banner__content\\">
      <!----> Chat Box
    </div>
    <!---->
  </div>
  <div class=\\"v-card v-theme--light v-card--density-default v-card--variant-flat d-flex flex-column-reverse overflow-y-auto\\" style=\\"height: 300px;\\">
    <!---->
    <div class=\\"v-card__loader\\">
      <div class=\\"v-progress-linear v-theme--light\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
        <!---->
        <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
        <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
          <div class=\\"v-progress-linear__indeterminate\\">
            <div class=\\"v-progress-linear__indeterminate long\\"></div>
            <div class=\\"v-progress-linear__indeterminate short\\"></div>
          </div>
        </transition-stub>
        <!---->
      </div>
    </div>
    <!---->
    <!---->
    <div class=\\"v-card-item text-justify text-caption\\">
      <!---->
      <div class=\\"v-card-item__content\\">
        <!---->
        <!---->
      </div>
      <!---->
    </div>
    <!---->
    <!----><span class=\\"v-card__underlay\\"></span>
  </div>
  <hr class=\\"v-divider v-theme--light\\" aria-orientation=\\"horizontal\\" role=\\"separator\\">
  <form class=\\"v-form d-flex\\" novalidate=\\"\\">
    <div class=\\"v-input v-input--horizontal v-input--density-compact v-text-field\\">
      <!---->
      <div class=\\"v-input__control\\">
        <div class=\\"v-field v-field--appended v-field--single-line v-field--variant-solo v-theme--light\\" role=\\"textbox\\">
          <div class=\\"v-field__overlay\\"></div>
          <div class=\\"v-field__loader\\">
            <div class=\\"v-progress-linear v-theme--light\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
              <!---->
              <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
              <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                <div class=\\"v-progress-linear__indeterminate\\">
                  <div class=\\"v-progress-linear__indeterminate long\\"></div>
                  <div class=\\"v-progress-linear__indeterminate short\\"></div>
                </div>
              </transition-stub>
              <!---->
            </div>
          </div>
          <!---->
          <div class=\\"v-field__field\\" data-no-activator=\\"\\">
            <!----><label class=\\"v-label v-field-label\\" for=\\"input-0\\">
              <!---->Message
            </label>
            <!----><input size=\\"1\\" type=\\"text\\" id=\\"input-0\\" aria-describedby=\\"input-0-messages\\" class=\\"v-field__input\\">
            <!---->
          </div>
          <!---->
          <div class=\\"v-field__append-inner\\">
            <!----><i class=\\"mdi-send mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable\\" role=\\"button\\" aria-hidden=\\"false\\" aria-label=\\"Message appended action\\"></i>
          </div>
          <div class=\\"v-field__outline\\">
            <!---->
            <!---->
          </div>
        </div>
      </div>
      <!---->
      <!---->
    </div>
  </form>
  <!---->
  <!----><span class=\\"v-card__underlay\\"></span>
</div>"
`;
